src/async_handle.rs

```diff
@@ -0,0 +1,78 @@
+use pin_project::pin_project;
+use std::future::Future;
+use std::panic::resume_unwind;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::thread;
+use thiserror::Error;
+use tokio::sync::oneshot::error::RecvError;
+use tokio::sync::oneshot::Receiver;
+
+/// Async handle for a blocking task running in a Rayon thread pool.
+///
+/// If the spawned task panics, `poll()` will propagate the panic.
+#[must_use]
+#[pin_project]
+#[derive(Debug)]
+pub struct AsyncHandle<T> {
```

Comment:
I would probably call this a rayon handle. Also, you shouldn't need to pin a oneshot channel as it is already Unpin.

Reply:
Good call to reference `Rayon` in the name. I went with `AsyncRayonHandle` to be extra clear.

Reply:
Also, thanks for the pinning tip - I am new to the concept of pinning so I was overcomplicating it :)